### ISSUE TITLE
Appending a ParamList object with a Parameterized object?

### ISSUE THREAD

**Issue by aboustati** [GPflow/GPflow#605]
Hello,

I am trying to append a `ParamList` object with a `Parameterized` object. I get the following error:
```
ValueError: Non parameter type cannot be appended to the list.
```

I found this a bit strange as I am able to initialise a `ParamList` object with a list of `Parameterized` objects, but cannot append it further. Is there a reason behind this behaviour or is it just a bug?

Thanks!

**Comment by aboustati** [comment]
In case this is not intentional behaviour, I propose the following fix:
```python
def __init__(self, list_of_params, trainable=True, name=None):
        super(ParamList, self).__init__(name=None)
        if not isinstance(list_of_params, list):
            raise ValueError('Not acceptable argument type at list_of_params.')
        self._trainable = trainable
        self._list = [self._valid_list_input(item, self._trainable)
                      for item in list_of_params]
        for index, item in enumerate(self._list):
            self._set_param(index, item)
```
```python
def append(self, item):
        item = self._valid_list_input(item, self._trainable)
        length = self.__len__()
        item.set_parent(self)
        item.set_name(self._item_name(length))
        self._list.append(item)
```
Please let me know if you'd like me to submit a pull request with these changes.

**Comment by st--** [comment]
Should be solved by #645
